prediction_market_agent/agents/think_thoroughly_agent/deploy.py

```diff
@@ -17,9 +20,17 @@ def load(self) -> None:
     def answer_binary_market(self, market: AgentMarket) -> Answer | None:
         return self.agent.answer_binary_market(market.question)
 
+    def before(self, market_type: MarketType) -> None:
+        self.agent.update_markets()
```

Comment:
Note that `update_markets` is called, so that we constantly update the embeddings present on Pinecone every time `think-thoroughly` runs.
I was a bit conservative (checks for markets opened in prev 7 days) to account for failed runs. When uploading to Pinecone, it checks if market already there (by ID), and only uploads if market not already present.
Net effect on average is one extra subgraph call per run, so should not affect the run too much.